File: challenges/2026-09-05-coprime-storage-bins/solution.cs
using System;
using System.Collections.Generic;

public class Solution
{
    public static int MinStorageBins(int[] batchCodes)
    {
        // TODO: implement
        return -1;
    }

    public static void Main(string[] args)
    {
        var tests = new List<(string Name, int[] BatchCodes, int Expected)>
        {
            ("example 1", new[] {6, 10, 15}, 3),
            ("example 2", new[] {4, 9, 25, 6}, 2),
            ("example 3", new[] {2, 4, 8, 16}, 4),
            ("edge: single sample", new[] {7}, 1),
            ("edge: chain conflict needs two bins", new[] {2, 3, 4}, 2),
        };

        foreach (var (name, batchCodes, expected) in tests)
        {
            int got = MinStorageBins(batchCodes);
            string status = got == expected ? "PASS" : "FAIL";
            Console.WriteLine($"[{status}] {name}: expected={expected} actual={got}");
        }
    }
}

File: challenges/2026-09-05-coprime-storage-bins/solution.go
package main

import "fmt"

func minStorageBins(batchCodes []int) int {
	// TODO: implement
	return -1
}

type testCase struct {
	name       string
	batchCodes []int
	expected   int
}

func main() {
	tests := []testCase{
		{"example 1", []int{6, 10, 15}, 3},
		{"example 2", []int{4, 9, 25, 6}, 2},
		{"example 3", []int{2, 4, 8, 16}, 4},
		{"edge: single sample", []int{7}, 1},
		{"edge: chain conflict needs two bins", []int{2, 3, 4}, 2},
	}

	for _, tc := range tests {
		got := minStorageBins(tc.batchCodes)
		status := "FAIL"
		if got == tc.expected {
			status = "PASS"
		}
		fmt.Printf("[%s] %s: expected=%d actual=%d\n", status, tc.name, tc.expected, got)
	}
}
